src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -229,7 +227,7 @@ public void SetThreadPrincipal(IPrincipal principal)
             lock (_forLock) {
                 // Check that principal has not been set previously.
                 if (_defaultPrincipal != null)
-                    throw new PolicyException(SR.AppDomain_Policy_PrincipalTwice);
+                    throw new SystemException(SR.AppDomain_Policy_PrincipalTwice);
```

Comment:
We could move it down if we thought it was useful but I'm not convinced it is for just this one case.